docs/milestone-1.md

```diff
@@ -0,0 +1,175 @@
+# Milestone 1 - kintsugi testnet
+
+This initial milestone provides simple middleware logic with minimal consensus client changes, simple networking, no validator authentication, and manual safety mechanism
+
+## Architecture
+
+### Block Proposal
+
+```sequence
+participant consensus
+participant mev_boost
+participant execution
+participant relays
+Title: Block Proposal
+Note over consensus: wait for allocated slot
+consensus->mev_boost: engine_forkchoiceUpdatedV1
+mev_boost->execution: engine_forkchoiceUpdatedV1
+mev_boost->relays: engine_forkchoiceUpdatedV1
+Note over mev_boost: begin polling
+mev_boost->relays: relay_getPayloadHeaderV1
+consensus->mev_boost: builder_getPayloadHeaderV1
+mev_boost->execution: engine_getPayloadV1
+Note over mev_boost: select best payload
+mev_boost-->consensus: builder_getPayloadHeaderV1 response
+Note over consensus: sign the block
+consensus->mev_boost: builder_proposeBlindedBlockV1
+Note over mev_boost: identify payload source
+mev_boost->relays: relay_proposeBlindedBlockV1
+Note over relays: validate signature
+relays-->mev_boost: relay_proposeBlindedBlockV1 response
+mev_boost-->consensus: builder_proposeBlindedBlockV1 response
+```
+
+## Specification
+
+1. mev-boost must be initialized with a list of (BLSPublicKey, IP address) pairs representing trusted relays.
+2. mev-boost must intercept [`engine_forkchoiceUpdatedV1`](#engine_forkchoiceupdatedv1) call from the BN -> EC and forward it to all connected relays to communicate `feeRecipient`.
+3. mev-boost must begin polling connected relays for their [`SignedMEVPayloadHeader`](#signedmevpayloadheader) using [`relay_getPayloadHeaderV1`](#relay_getpayloadheaderv1) requests.
+4. mev-boost must verify the returned [`SignedMEVPayloadHeader`](#signedmevpayloadheader) signature matches the BLS key associated with the IP of the relay and has a matching `payloadId`.
+5. upon receiving a [`builder_getPayloadHeaderV1`](#builder_getpayloadheaderv1) request from the BN, mev-boost must return the [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayloadheader) with the highest associated `feeRecipientBalance`. If no eligible payload is received from a relay, mev-boost must request and return a payload from the local execution client using [`engine_getPayloadV1`](#engine_getpayloadv1).
```

Comment:
What do we think of the non-middleware solution proposed here:
https://hackmd.io/@paulhauner/H1XifIQ_t#Is-mev_boost-Middleware

Then we can remove `If no eligible payload is received from a relay, mev-boost must request and return a payload from the local execution client using [`engine_getPayloadV1`](#engine_getpayloadv1)...` and remove `engine_getPayloadV1` call below

Reply:
we would like to move in that direction for our communication with the consensus layer. Take a look at @metachris' exploration in #53.
Also please take a look at the updated diagram in #57.